pkg/cloudprovider/vsphere/config.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vsphere
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"gopkg.in/gcfg.v1"
+)
+
+// CPIConfig is used to read and store information (related only to the CPI) from the cloud configuration file
+type CPIConfig struct {
+	CPI struct {
+		// address options for each network type (internal and/or external) are mutually inclusive
+		// e.g., if you specify "internal-network-name" and "internal-network-subnet-cidr"
+		// only the addresses that match BOTH will match
+
+		// VirtualMachine network interfaces backed by the following vSphere Networks
+		// will be used in respective status.addresses fields.
+		InternalNetworkName []string `gcfg:"internal-network-name"`
```

Comment:
How do we feel about making these accept just 1 entry and expanding this to a list in the future?

Reply:
I'm skeptical about use-cases involving two networks per network type, if we have valid use-cases that work when specifying more than 1 InternalIP for example, then I think we should consider this. I personally have not seen this work on vSphere (yet).

Reply:
I've made a separate commit to roll back changes to a single possible network. I've pondered upon the use cases, and I've found a single justification for multiple network names. This CPI implementation supports **multiple** vCenters. There could be a slim chance of a having a *single* L2 segment (e.g., VLAN) under different *network names* in vCenters. 

But if we are moving away from network names as a selector, then this isn't valid anymore. The L3 IP subnet will stay the same regardless of any vCenter configurations.

Reply:
sounds good to me :)